local-interchaintest/examples/token_swap.rs

```diff
@@ -0,0 +1,390 @@
+        .with_label("swap")
+        .with_actions_config(
+            AtomicActionsConfigBuilder::new()
+                .with_action(
+                    AtomicActionBuilder::new()
+                        .with_contract_address(&splitter1.address)
+                        .with_message_details(MessageDetails {
+                            message_type: MessageType::CosmwasmExecuteMsg,
+                            message: Message {
+                                name: "process_action".to_string(),
+                                params_restrictions: Some(vec![ParamRestriction::MustBeIncluded(
+                                    vec!["process_action".to_string(), "split".to_string()],
+                                )]),
+                            },
+                        })
+                        .build(),
+                )
+                .with_action(
+                    AtomicActionBuilder::new()
+                        .with_contract_address(&splitter2.address)
+                        .with_message_details(MessageDetails {
+                            message_type: MessageType::CosmwasmExecuteMsg,
+                            message: Message {
+                                name: "process_action".to_string(),
+                                params_restrictions: Some(vec![ParamRestriction::MustBeIncluded(
+                                    vec!["process_action".to_string(), "split".to_string()],
+                                )]),
+                            },
+                        })
+                        .build(),
+                )
+                .build(),
+        )
+        .build()];
+
+    info!("Creating swap authorization...");
+    let create_authorization = valence_authorization_utils::msg::ExecuteMsg::PermissionedAction(
+        valence_authorization_utils::msg::PermissionedMsg::CreateAuthorizations { authorizations },
+    );
+
+    contract_execute(
+        test_ctx
+            .get_request_builder()
+            .get_request_builder(NEUTRON_CHAIN_NAME),
+        &authorization_contract_address,
+        DEFAULT_KEY,
+        &serde_json::to_string(&create_authorization).unwrap(),
+        GAS_FLAGS,
+    )
+    .unwrap();
+
+    info!("Swap authorization created!");
+    std::thread::sleep(std::time::Duration::from_secs(3));
+
+    info!("Send the messages to the authorization contract...");
+    let binary = Binary::from(
+        serde_json::to_vec(&valence_service_utils::msg::ExecuteMsg::<
+            ActionMsgs,
+            OptionalServiceConfig,
+        >::ProcessAction(ActionMsgs::Split {}))
+        .unwrap(),
+    );
+    let message = ProcessorMessage::CosmwasmExecuteMsg { msg: binary };
+
+    let send_msg = valence_authorization_utils::msg::ExecuteMsg::PermissionlessAction(
+        valence_authorization_utils::msg::PermissionlessMsg::SendMsgs {
+            label: "swap".to_string(),
+            messages: vec![message.clone(), message],
```

Comment:
Why are 2 messages being sent?

Reply:
One for each splitter. The workflow is Account1->Splitter1->Account2 and Account2 ->Splitter2->Account1